cpp/src/arrow/util/logging.cc

```diff
@@ -113,10 +115,14 @@ void ArrowLog::StartArrowLog(const std::string& app_name,
                              ArrowLogLevel severity_threshold,
                              const std::string& log_dir) {
   severity_threshold_ = severity_threshold;
-  app_name_.reset(new std::string(app_name.c_str()));
+  // In InitGoogleLogging, it simply keeps the pointer.
+  // We need to make sure the app name passed to InitGoogleLogging exist.
+  // We should avoid using static string is a dynamic lib.
+  static std::unique_ptr<std::string> app_name_;
```

Comment:
I'm not sure I understand why there's one declaration (`log_dir_`) at the top level and another (`app_name_`) in this function.

Reply:
`app_name_` is used in `google::InitGoogleLogging` but not used in other places. `log_dir_ ` is used to set the log file directory and will be used in `ShutDownArrowLog`, so `log_dir_` is defined as a global variable.